services/src/floodlight-service/floodlight-api/src/main/java/org/openkilda/floodlight/api/request/IngressFlowSegmentBase.java

```diff
@@ -45,14 +45,19 @@
     @JsonProperty("remove_customer_port_shared_catch_rule")
     protected final boolean removeCustomerPortSharedCatchRule;
 
+    @JsonProperty("remove_customer_port_shared_lldp_catch_rule")
+    protected final boolean removeCustomerPortSharedLldpCatchRule;
+
     IngressFlowSegmentBase(
             MessageContext context, UUID commandId, FlowSegmentMetadata metadata, @NonNull FlowEndpoint endpoint,
-            MeterConfig meterConfig, @NonNull SwitchId egressSwitchId, boolean removeCustomerPortSharedCatchRule) {
+            MeterConfig meterConfig, @NonNull SwitchId egressSwitchId, boolean removeCustomerPortSharedCatchRule,
+            boolean removeCustomerPortSharedLldpCatchRule) {
```

Comment:
I'd recommend to encapsulate such flags into a separate structure and re-use it in corresponding requests/factories/commands. This will simplify future changes - affecting the only structure, not dozens of classes.

Reply:
fixed